### PR TITLE
Layers: Multi project inheritance #8233

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentContentSynchronizer.java [hunk 1]
```diff
@@ -84,23 +84,36 @@ public void sync( final ContentSyncParams params )
...
+        else
+        {
+            final List<ContentToSync> contentsToSync;
+            try
             {
-                this.doSyncWithChildren( contentsToSync, targetContexts );
+                contentsToSync = createContentsToSync( params.getContentIds(), sourceContexts, targetContexts );
             }
-            else
+            catch ( Exception e )
```
Comment (rymsha) [inline modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentContentSynchronizer.java:102]: why?

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentContentSynchronizer.java [hunk 2]
```diff
@@ -125,7 +149,14 @@ private void doSyncWithChildren( final List<ContentToSync> sourceContents, final
         final List<ContentToSync> contentsToSync = sourceContents.stream().filter( sourceContent -> {
 
             final Content root = sourceContent.getSourceContext().callWith( () -> contentService.getByPath( ContentPath.ROOT ) );
-            return !root.getId().equals( sourceContent.getSourceContent().getId() );
+            try
+            {
+                return !root.getId().equals( sourceContent.getSourceContent().getId() );
+            }
+            catch ( Exception e )
```
Comment (rymsha) [inline modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentContentSynchronizer.java:156]: why?

File: modules/core/core-api/src/main/java/com/enonic/xp/project/CreateProjectParams.java [hunk 3]
```diff
@@ -55,9 +59,15 @@ public String getDescription()
         return description;
     }
 
+    public List<ProjectName> getParents()
+    {
+        return parents;
+    }
+
+    @Deprecated
```
Comment (rymsha) [inline modules/core/core-api/src/main/java/com/enonic/xp/project/CreateProjectParams.java:67]: parent is a first item from parents. Don't deprecate

File: modules/lib/lib-project/src/test/resources/test/CreateProjectHandlerTest.js [hunk 4]
```diff
@@ -35,30 +35,31 @@ exports.createProject = function () {
...
 
-var createProjectWithParentExpected = {
+var createProjectWithOneParentExpected = {
     id: 'myproject',
     displayName: 'project display name',
     description: 'project description',
-    parent: 'testparent',
     language: 'no',
+    parents: ['testparent'],
+    parent: 'testparent', //TODO: discuss backward compatibility
```
Comment (rymsha) [inline modules/lib/lib-project/src/test/resources/test/CreateProjectHandlerTest.js:44]: parent is a first item from parents. Don't deprecate

**Review by rymsha — APPROVED** [review]

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentContentSynchronizer.java [hunk 5]
```diff
@@ -110,8 +123,19 @@ public void sync( final ContentEventsSyncParams params )
     {
         final Map<NodePath, Context> sourceContexts = initContexts( params.getSourceProject().getRepoId() );
         final Map<NodePath, Context> targetContexts = initContexts( params.getTargetProject().getRepoId() );
+
         final List<ContentToSync> contents = createContentsToSync( params.getContentIds(), sourceContexts, targetContexts );
 
+        if ( contents.isEmpty() )
+        {
+            throw new IllegalArgumentException( ( String.format( "nothing to sync, content ids: [%s], source: [%s], target: [%s]",
```
Comment (anatol-sialitski) [inline modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentContentSynchronizer.java:131]: Why it should failed here? Just log that nothing to sync or skip.

File: modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/mapper/ProjectMapper.java [hunk 6]
```diff
@@ -84,6 +84,19 @@ private void serializeSiteConfigs( final MapGenerator gen )
...
+    private void serializeParents( final MapGenerator gen )
+    {
+        if ( !project.getParents().isEmpty() )
+        {
+            gen.array( "parents" );
+            project.getParents().forEach( gen::value );
+            gen.end();
+
+            gen.value( "parent", project.getParents().get( 0 ) ); // deprecated
+        }
```
Comment (anatol-sialitski) [inline modules/lib/lib-project/src/main/java/com/enonic/xp/lib/project/mapper/ProjectMapper.java:96]: Maybe `parent` should be left here for backward compatibility?

**Review by anatol-sialitski — APPROVED** [review]

**Pull request merged**